cost/azure/reserved_instances/recommendations/azure_reserved_instance_recommendations.pt

```diff
@@ -394,31 +394,86 @@ script "js_azure_subscriptions_filtered", type: "javascript" do
 EOS
 end
 
+# Only request the resource groups in each subscription if we're doing "Resource Group" scope
+datasource "ds_azure_resource_groups_requests" do
+  run_script $js_azure_resource_groups_requests, $ds_azure_subscriptions_filtered, $param_scope
+end
+
+script "js_azure_resource_groups_requests", type: "javascript" do
+  parameters "ds_azure_subscriptions_filtered", "param_scope"
+  result "result"
+  code <<-EOS
+  result = []
+  if (param_scope == "Resource Group") { result = ds_azure_subscriptions_filtered }
+EOS
+end
+
+datasource "ds_azure_resource_groups" do
+  iterate $ds_azure_resource_groups_requests
+  request do
+    auth $auth_azure
+    pagination $pagination_azure
+    host $param_azure_endpoint
+    path join(["/subscriptions/", val(iter_item, "id"), "/resourcegroups"])
+    query "api-version", "2021-04-01"
+    header "User-Agent", "RS Policies"
+    ignore_status [400, 403, 404]
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "resourceGroup", jmes_path(col_item, "name")
+      field "id", val(iter_item, "id")
+      field "name", val(iter_item, "name")
+    end
+  end
+end
+
 datasource "ds_ri_requests" do
-  run_script $js_ri_requests, $ds_azure_subscriptions_filtered, $ds_service_table, $param_service, $param_scope
+  run_script $js_ri_requests, $ds_azure_subscriptions_filtered, $ds_azure_resource_groups, $ds_service_table, $param_service, $param_scope
 end
 
 script "js_ri_requests", type: "javascript" do
-  parameters "ds_azure_subscriptions_filtered", "ds_service_table", "param_service", "param_scope"
+  parameters "ds_azure_subscriptions_filtered", "ds_azure_resource_groups", "ds_service_table", "param_service", "param_scope"
   result "result"
   code <<-EOS
   result = []
 
   if (param_scope == "Shared") {
+    // Remove subscriptions that don't contain actual resources
+    good_subs = _.filter(ds_azure_subscriptions_filtered, function(sub) {
+      return sub['name'].toLowerCase().indexOf("visual studio") == -1 && sub['name'].indexOf("365") == -1
```

Comment:
tested in customer org where the issue was identified and the fix works as expected